### PR TITLE
Station data

### PR DESCRIPTION
**Pull request by aaron-kaplan**
Accept observations in station data format.

### CODE REVIEW

**Comment by aaron-kaplan** [comment]
@xchourio I have rebased this branch, including the commit you added.

Your change causes pycpt-station.ipynb to use the predictor domain instead of the predictand domain for the skill and forecast plots. The first screenshot is from the commit before yours, the second from yours.

Screenshot from 2024-10-17 12-12-18
Screenshot from 2024-10-17 12-03-37

**Comment by xchourio** [comment]
> @xchourio I have rebased this branch, including the commit you added.
> 
> Your change causes pycpt-station.ipynb to use the predictor domain instead of the predictand domain for the skill and forecast plots. The first screenshot is from the commit before yours, the second from yours.
> 
> Screenshot from 2024-10-17 12-12-18 Screenshot from 2024-10-17 12-03-37

By default, if it doesn’t receive a domain, it adjusts 5 degrees around the edge of the station coordinates; otherwise, it becomes so tight that it doesn’t display well (at least that was the case with the Chile stations notebook). To plot the domain, the predictand’s domain must be passed as a parameter.

**Comment by aaron-kaplan** [comment]
How about something like 2% of the total height/width (experiment to find a % that looks good) instead of 5 degrees? 5 degrees is a lot at this scale. And I think it's important to pick a reasonable default rather than making the user figure out what to use.

**Comment by aaron-kaplan** [comment]
I'm going to try that now, and will push it for your review if it seems to work.

**Comment by aaron-kaplan** [comment]
Having tried it, now I understand the problem more clearly. Adding 10% margin to the New York State map looks pretty good, but Chile is so narrow that adding an extra 10% to the width still isn't enough.

But since Chile is a rather unusual case, I suggest that we go with 10% instead of 5 degrees as the default, and use an explicit domain for Chile.

@xchourio this whole branch is now ready for review. I know it's a lot; hopefully the new tests give us more confidence.

File: pycpt/src/cptextras/view.py [hunk 1]
```diff
@@ -5,43 +5,40 @@
...
+def view_coords_stations(domain,vals):
+    step=0.1
+    horizontal = len( np.arange(vals['X'].values.min(), vals['X'].values.max() + step, step) )
+    vertical = len( np.arange(vals['Y'].values.min(), vals['Y'].values.max() + step, step) )
+    
+    if horizontal>=vertical:
+       factor_percent_h=0.10
+       factor_percent_v=0.20
+    else:
+	   #50% seems like a lot in general, but it manages to cover Chile's borders well.
```
Comment (aaron-kaplan) [inline pycpt/src/cptextras/view.py:18]: @xchourio I think 50% would be excessive in the vast majority of cases, even countries that are vertically oriented. I think the shape of Chile is so unusual that we should just treat it as a special case, by using the explicit `domain` kwarg that you added.

I did that yesterday in https://github.com/iri-pycpt/notebooks/pull/15/commits/a993a682ff25d994721a39a58bb3c2db8ec866e8, but you didn't react to it, so maybe you didn't see it.

Comment (aaron-kaplan) [reply pycpt/src/cptextras/view.py:18]: BTW what's special about Chile is not only that it's tall and narrow, but also that there are no stations in the southwest coastal area, which leads to that part of the country getting cut off by the default cropping.

Comment (xchourio) [reply pycpt/src/cptextras/view.py]: > BTW what's special about Chile is not only that it's tall and narrow, but also that there are no stations in the southwest coastal area, which leads to that part of the country getting cut off by the default cropping.

I revoked my last changes.

**Comment by aaron-kaplan** [comment]
OK. No harm done, but in future, please don't `git push --force` to other people's branches. Only the owner of a branch should force push, otherwise there is a risk of work being lost: if I pushed a commit after your last pull but before your force-push, my commit would be lost.

(master is the one branch that has no specific owner, and for that reason, we should never force-push to master.)

You can retract a commit without force-pushing by using `git revert <commit-hash>`.

@xchourio please review this branch, using the `station` branch of `notebooks` for testing.

**Comment by aaron-kaplan** [comment]
Added two more commits that fix overlapping titles in the flex forecast figure.

**Comment by aaron-kaplan** [comment]
@xchourio see my latest commits. I'm now adjusting the X and Y coordinates in the functions that read CPT's outputs, so everywhere else they should match the originals. That way we don't have to hunt down every place where they might differ. I reversed the changes you made in https://github.com/iri-pycpt/pycpt/commit/3582390fdb509ea50d210d2360a6e656b367d4ba because I think they should no longer be necessary, but I never managed to reproduce the problem that led to those changes, so I'm not positive. Please test it and confirm.

The most recent commit is unrelated to that. Prior to that commit, a multi-month season was represented in the generated cptv10.tsv as e.g. `1996-12/1997-02`, but a one-month season was represented as e.g. `1996-12-15:T00:00.000000000`. That mostly worked, but there are situations where it causes problems.

**Comment by xchourio** [comment]
> @xchourio see my latest commits. I'm now adjusting the X and Y coordinates in the functions that read CPT's outputs, so everywhere else they should match the originals. That way we don't have to hunt down every place where they might differ. I reversed the changes you made in [3582390](https://github.com/iri-pycpt/pycpt/commit/3582390fdb509ea50d210d2360a6e656b367d4ba) because I think they should no longer be necessary, but I never managed to reproduce the problem that led to those changes, so I'm not positive. Please test it and confirm.
> 
> The most recent commit is unrelated to that. Prior to that commit, a multi-month season was represented in the generated cptv10.tsv as e.g. `1996-12/1997-02`, but a one-month season was represented as e.g. `1996-12-15:T00:00.000000000`. That mostly worked, but there are situations where it causes problems. Now we generate the correct representation, e.g. `1996-12`.

That has worked perfectly with all tests, including grid, stations, None, and Anomaly, as well as precip and temps.

I ran the tests with this notebook, which already includes options to detect whether it should plot a point or grid (final blocks for the flex format):  
https://github.com/iri-pycpt/notebooks/commit/9d99b96f6950ad7ee31da36bf344275f826eef63#diff-4e69287be15b01551b893a62b8f4a7cdea899e9b786df6c22d5d89143cf6ea8b

**Review by xchourio — APPROVED** [review]

**Pull request merged**